hash.c

```diff
@@ -3278,21 +3278,35 @@ each_value_i(VALUE key, VALUE value, VALUE _)
 
 /*
  *  call-seq:
- *     hsh.each_value {| value | block } -> hsh
- *     hsh.each_value                    -> an_enumerator
+ *    hash.each_value {|value| ... } -> self
+ *    hash.each_value -> new_enumerator
  *
- *  Calls <i>block</i> once for each key in <i>hsh</i>, passing the
- *  value as a parameter.
- *
- *  If no block is given, an enumerator is returned instead.
+ *  Calls the given block with each value; returns +self+:
+ *    h = {foo: 0, bar: 1, baz: 2}
+ *    h1 = h.each_value {|value| puts value }
+ *    h1 # => {:foo=>0, :bar=>1, :baz=>2}
+ *    h1.equal?(h) # => true # Identity check
```

Comment:
I understand why this is logical, but I'm not sure what it adds. It seems like a distraction?

Reply:
Shows that it's +self+ that's returned.  (But elsewhere I did not show that the return is a new Hash).